### PR TITLE
feat: Implementation of CRUD API for Access management APIs - Add rule validations

### PR DESCRIPTION
**Pull request by istvan-nagy-epam**
## Description
- Adds custom validator implementations
- Configures validation groups using JSR-303 annotations
- Adds tests to verify that validation is working
- Fixes internal logic of policy values to allow validation rules to enforce validation instead of runtime exceptions

Updates #286 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: access-control-service-sql-impl/pom.xml [hunk 1]
```diff
@@ -92,6 +100,23 @@
...
       </dependency>
+      <dependency>
+         <groupId>org.mockito</groupId>
+         <artifactId>mockito-core</artifactId>
+         <scope>test</scope>
+      </dependency>
+      <dependency>
+         <groupId>org.hibernate.validator</groupId>
+         <artifactId>hibernate-validator</artifactId>
+         <version>8.0.1.Final</version>
```
Comment (tunacicek) [inline access-control-service-sql-impl/pom.xml:111]: Please move the version to the parent pom.

File: access-control-service-sql-impl/pom.xml [hunk 2]
```diff
@@ -92,6 +100,23 @@
...
+         <scope>test</scope>
+      </dependency>
+      <dependency>
+         <groupId>org.apache.tomcat.embed</groupId>
+         <artifactId>tomcat-embed-el</artifactId>
+         <version>10.1.18</version>
```
Comment (tunacicek) [inline access-control-service-sql-impl/pom.xml:117]: Please move the version to the parent pom.

File: access-control-service-sql-impl/pom.xml (hunk 1 above)
Comment (istvan-nagy-epam) [reply access-control-service-sql-impl/pom.xml]: Done, thank you!

File: access-control-service-sql-impl/pom.xml (hunk 2 above)
Comment (istvan-nagy-epam) [reply access-control-service-sql-impl/pom.xml]: Done, thank you!

File: access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/validation/AccessRulePolicyValueValidator.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+         boolean validatedMultiValues = validateMultiValues( accessRulePolicyValue, constraintValidatorContext );
+         valid = valid && validatedMultiValues;
+      }
+      if ( !valid ) {
+         constraintValidatorContext.buildConstraintViolationWithTemplate( message ).addConstraintViolation();
+      }
+      return valid;
+   }
+
+   private static boolean validateSingleValue( AccessRulePolicyValue accessRulePolicyValue, ConstraintValidatorContext constraintValidatorContext ) {
```
Comment (tunacicek) [inline access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/validation/AccessRulePolicyValueValidator.java:70]: Is static required here?

Comment (istvan-nagy-epam) [reply access-control-service-sql-impl/src/main/java/org/eclipse/tractusx/semantics/accesscontrol/sql/validation/AccessRulePolicyValueValidator.java:70]: No, sorry! Thank you for noticing!

**Pull request merged**